File: python/triangulo.py
#!/usr/bin python
def triangulo():
	texto = open("../triangle.txt", 'r').read()

	lineas = filter(None, texto.split('\n'))
	triangulo = [[]]

	for i in range(len(lineas)):
		items = lineas[i].split(' ')
		triangulo.append(items)
	
	triangulo.pop(0)

	for j in reversed(range(0, len(triangulo) - 1)):
		for k in range(0, j + 1):
			triangulo[j][k] = int(triangulo[j][k]) + int(max(triangulo[j+1][k], triangulo[j+1][k+1]))

	print triangulo[0][0]
	return

triangulo()

File: ruby/triangulo.rb
def triangulo()
	texto = File.new("../triangle.txt", "r")
	triangulo = [[]]

	while (linea = texto.gets)
		triangulo << linea.split(' ')
	end

	triangulo.shift

	rango = triangulo.length - 2..0

	(rango.first).downto(rango.last).each { |i| 
		for j in 0..i + 1
			triangulo[i][j] = triangulo[i][j].to_i + [triangulo[i+1][j].to_i, triangulo[i+1][j+1].to_i].max
		end
	}
	puts triangulo[0][0]
end

triangulo()